packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -175,9 +175,11 @@ export class Outbox {
 		if (
 			batch.content.length === 0 ||
 			this.params.config.compressionOptions === undefined ||
-			this.params.config.compressionOptions.minimumBatchSizeInBytes > batch.contentSizeInBytes
+			this.params.config.compressionOptions.minimumBatchSizeInBytes >
+				batch.contentSizeInBytes ||
+			this.params.containerContext.submitBatchFn === undefined
```

Comment:
I assume this gives us only partial safety, as there are loader versions who have submitBatchFn, but do not propagate compression flag, right?

Reply:
This is just a performance improvement. Currently, if submitBatchFn === undefined, we won't be sending the compressed ops, we'll be sending the deserialized content:

```
				this.params.containerContext.submitFn(
					MessageType.Operation,
					message.deserializedContent,
					true, // batch
					message.metadata,
				);
```
so there's no need to even do compression in the first place as it will be discarded anyway. Additionally, compression without sending it pollutes the log and gives us false impressions about the feature usage..